components/encoder/single/single_encoder.go

```diff
@@ -208,7 +208,8 @@ func (e *Encoder) start(ctx context.Context, b board.Board) {
 					atomic.AddInt64(&e.position, dir)
 				}
 			} else {
-				e.logger.CDebug(ctx, "received tick for encoder that isn't connected to a motor; ignoring")
```

Comment:
Do we have a place to warn the user that the encoder cannot tell which direction it is going and so will always tick up when it receives a tick?

Reply:
I will add a debug log

Reply:
huh, I don't see where directional awareness is attached in this driver. It must be in gpio motor's setup function.  Nothing to do now, just noting this for myself and others.

Reply:
A debug log here is fine.

Reply:
yeah its set in GPIO motor, I need to make a change to the PR to remove the directional awareness if the encoder is taken out of the motor config. tested putting on the motor yesterday but not taking it back off again :/

Reply:
Okay actually this is same behavior that the encoder has before this change so I guess it depends if we want to change the behavior. Currently if the encoder is taken out of the motor config, we never remove the directional awareness from the encoder and the `e.m` still points to the motor. Do we want to change it to remove the awareness and tick in the positive direction? It would require implementing `reconfigure` in the gpio motor so might be kind of a large change, is there a reason that gpio motor is currently `AlwaysRebuild` ?

Reply:
Yes, I do not want changing pins or state on the motor to ever not require a stop, close and rebuild of everything. It is a complex configuration schema and it is not worth the effort to implement Reconfigure and figure out the edge cases. A single encoder needing awareness of the direction it's going is not reason enough to implement reconfiguring in the motor. 

It's fine to keep the behaviour the same as it was before. This is just a pattern we'll have to deal with eventually. If we ever modularize these two components, the code we have in rdk will not work, as we return mtoro and encoder clients  (`motor.Motor`, `encoder.Encoder`) between modules - the the rdk we can return specific structs and type check them (`EncodedMotor` `singleencoder`).

Just confirming, can we tick forward in some cases right now? Should we `alwaysRebuild` this resource? I think a resource parent commands the child resources to rebuild, but that's a question to ask netcode.

Reply:
Okay sounds good will keep it the same. We only will go the `else` case if no motor has ever added the encoder to its config. I don't think always rebuilding the encoder will make a difference, as the direction awareness is only set by the motor.

Reply:
One thing to note - this pattern ONLY works for gpio motor, we have no way to attach directional awareness to single encoder from any other motor, so this is an antipattern that we will have to remove anyway, and we should not expand on it. 

If I attach the single encoder to a roboclaw right now, it will not be able to figure out its direction of movement.

Reply:
yeah great point